lib/ansible/modules/web_infrastructure/ansible_tower/tower_credential.py

```diff
@@ -257,14 +266,18 @@ def main():
                 team = team_res.get(name=module.params.get('team'))
                 params['team'] = team['id']
 
-            if module.params.get('ssh_key_data'):
-                filename = module.params.get('ssh_key_data')
-                if not os.path.exists(filename):
-                    module.fail_json(msg='file not found: %s' % filename)
-                if os.path.isdir(filename):
-                    module.fail_json(msg='attempted to read contents of directory: %s' % filename)
-                with open(filename, 'rb') as f:
-                    module.params['ssh_key_data'] = to_text(f.read())
+            ssh_key_data = module.params.get('ssh_key_data')
+            if ssh_key_data:
+                if ssh_key_data.startswith('-----'):
```

Comment:
It might be better to use a similar regex that awx itself uses:

https://github.com/ansible/awx/blob/devel/awx/main/validators.py#L51

Maybe just validate the format of the first line?

```python
if re.search(r'^(?P<dashes>-{4,}) *BEGIN (?P<type>[A-Z ]+?) *(?P=dashes)', ssh_key_data):
```

Reply:
I wonder if ansible itself has something for this in its `authorized_keys` module.